dask_expr/expr.py

```diff
@@ -1671,14 +1708,33 @@ def optimize(expr: Expr, fuse: bool = True) -> Expr:
     simplify
     optimize_blockwise_fusion
     """
-    expr = expr.simplify()
+    expr = expr.simplify().lower()
 
     if fuse:
         expr = optimize_blockwise_fusion(expr)
 
     return expr
 
 
+def non_blockwise_ancestors(expr):
+    """Traverse through tree to find ancestors that are not blockwise or are IO"""
+    stack = [expr]
+    while stack:
+        e = stack.pop()
+        if isinstance(e, IO):
+            yield e
+        elif isinstance(e, Blockwise):
+            stack.extend(e.dependencies())
+        else:
+            yield e
```

Comment:
Maybe broadcastable expressions should also be excluded here

`df.x + a.sum()` and `df.y + b.sum()` should still be co-aligned even if `a` and `b` are quite different.  (this would make a good test somewhere probably).

Reply:
That's a good point, better excluding them here than in AlignDivisions. Added this to #188